modmail/dispatcher.py

```
@@ -0,0 +1,167 @@
+        If you have accidentally added a handler twice to an event, this will only remove one instance.
+        """
+        if not event_names:
+            event_names = self.handlers.keys()
+
+        for event_name in event_names:
+            if event_name not in self.handlers:
+                logger.exception(
+                    "Attempted to unregister handler %r from event name %s, which wasn't registered.",
+                    func,
+                    event_name,
+                )
+                continue
+
+            if func in self.handlers[event_name]:
+                self.handlers[event_name].remove(func)
+
+            if func in self.blocking_handlers[event_name]:
+                # blocking handlers are two separate lists because it makes searching for items
+                # (and thus bisect) a hundred times easier. But that does mean we have to keep them in sync.
+                index = self.blocking_handlers[event_name].index(func)
+                del self.blocking_handlers[event_name][index]
+                del self.blocking_priorities[event_name][index]
+
+    async def dispatch(self, event_name: str, *args, **kwargs) -> None:
+        """
+        Trigger dispatch of an event, passing args directly to each handler.
+
+        Beware passing mutable args--previous handlers, if misbehaving, can mutate them.
+        """
+        if event_name not in self.blocking_handlers:
+            logger.exception(
+                "Unregistered event '%s' was dispatched to no handlers with data: %r %r",
+                event_name,
+                args,
+                kwargs,
+            )
+            self.blocking_handlers[event_name] = []
+            self.handlers[event_name] = []
+            self.blocking_priorities[event_name] = []
+
+        for handler in self.blocking_handlers[event_name]:
+            if await handler(*args, **kwargs):
+                return
```

Why does it return here with None? If its a blocking handler, it should only be blocked if the awaitable returns a Nonzero value.

This function should return the error code so the caller *knows* if the checks failed or not.

---

We return none here in order to cancel further dispatch of the vent